### PR TITLE
Pin k8s to < 28.1.0

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

28.1.0 broke us (maybe). Pinning for now.

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by schrockn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16605** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16605" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16605?utm_source=stack-comment).

File: python_modules/libraries/dagster-k8s/setup.py [hunk 1]
```diff
@@ -32,6 +32,6 @@ def get_version() -> str:
         "Operating System :: OS Independent",
     ],
     packages=find_packages(exclude=["dagster_k8s_tests*"]),
-    install_requires=[f"dagster{pin}", "kubernetes"],
+    install_requires=[f"dagster{pin}", "kubernetes<28.1.0"],
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-k8s/setup.py:35]: didn't spot anything in https://github.com/kubernetes-client/python/blob/release-28.0/CHANGELOG.md that stood out

don't love adding user effecting pins like this without a good understanding of what broke 

could consider pinning tests specifically like in `integration_tests/test_suites/k8s-test-suite/tox.ini` 

would be good to leave a comment next to the pin explaining why its pinned

Comment (schrockn) [reply python_modules/libraries/dagster-k8s/setup.py:35]: This did not appear to resolve the issue anyway. https://buildkite.com/dagster/dagster/builds/66435#018aaa08-fb11-43c5-a1e9-8c1239955284

**Review by alangenfeld — APPROVED** [review]
seems like it might be it though 
just to record what failed 
https://buildkite.com/dagster/dagster/builds/66389
https://buildkite.com/dagster/dagster/builds/66399
https://buildkite.com/dagster/dagster/builds/66402
https://buildkite.com/dagster/dagster/builds/66407

**Pull request closed without merging**